web/src/server/api/routers/generations/db/getAllGenerationsSqlQuery.ts

```diff
@@ -118,12 +120,11 @@ export async function getAllGenerations({
         p.name as "promptName",
         p.version as "promptVersion"
       FROM observations_view o
-      JOIN traces t ON t.id = o.trace_id AND t.project_id = o.project_id
+      JOIN traces t ON t.id = o.trace_id AND t.project_id = ${input.projectId}
```

Comment:
FYI: i dont think this change was necessary as we filter observations by projectId

Reply:
agree, I thought this makes it more clear and potentially only increases performance